UIKitTests/UIProgressViewTests.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  UIProgressViewTests.swift
+//  UIKitTests
+//
+//  Created by Michael Knoch on 26.06.18.
+//  Copyright © 2018 flowkey. All rights reserved.
+//
+
+import XCTest
+@testable import UIKit
+
+class UIProgressViewTests: XCTestCase {
+    var progressView: UIProgressView!
+
+    override func setUp() {
+        progressView = UIProgressView()
+        progressView.frame = CGRect(x: 0, y: 0, width: 100, height: 100)
+    }
+
+    func testCanSetProgressWithoutAnimation() {
+        let newProgress: Float = 0.5
+        progressView.setProgress(newProgress, animated: false)
+
+        XCTAssertEqual(progressView.progress, newProgress)
+        XCTAssertNil(progressView.progressLayer._presentation)
+    }
+
+    func testCanSetProgressWithAnimation() {
+        let newProgress: Float = 0.5
+        progressView.setProgress(newProgress, animated: true)
+        progressView.progressLayer.hasBeenRenderedInThisPartOfOverallLayerHierarchy = true
+        progressView.layoutSubviews()
+
+        XCTAssertEqual(progressView.progress, newProgress)
+        XCTAssertNotNil(progressView.progressLayer._presentation)
```

Comment:
we should always use the public API so we can test iOS too (_presentation is not public)